src/workarounds/forwarddiff_rules.jl

```diff
@@ -218,7 +218,19 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
     response.verbose && println("Solving response problem")
     δresults = ntuple(ForwardDiff.npartials(T)) do α
         δHextψ = [ForwardDiff.partials.(δHextψk, α) for δHextψk in Hψ_dual]
-        solve_ΩplusK_split(scfres, -δHextψ; tol=scfres.norm_Δρ, response.verbose)
+        # compute tolerance for the sternheimer solver as the maximum residual
+        # on the occupied eigenvectors, because asking the sternheimer solver to
+        # converge below this tolerance can lead to oversolving issues
+        residual_norms = scfres.diagonalization[1].residual_norms
+        # first, select only fully converged eigenvectors above the occupation
+        # threshold
+        mask_occ   = map(occk -> isless.(scfres.occupation_threshold, occk),
+                         scfres.occupation)
+        residual_norms_occ = [residual_norms[ik][maskk]
+                              for (ik, maskk) in enumerate(mask_occ)]
+        tol_sternheimer = maximum(maximum.(residual_norms_occ))
+        solve_ΩplusK_split(scfres, -δHextψ; tol=scfres.norm_Δρ,
+                           tol_sternheimer, response.verbose)
```

Comment:
`tol_sternheimer` should at most be `tol`, so you have to adapt tol as well.

Also this is a lot of repeated code with things that are also done in `solve_ΩplusK_split(scfres)`. I think that should be refactored.

Reply:
Ok, so we should just take `tol=scfres.norm_Δρ` for both the Dyson and the Sternheimer ?

Reply:
I'm more tempted to have the maximal residual be both `tol` and `tol_sternheimer`. See the discussion in #742.